apps/web/src/components/MintLinkButton.tsx

```diff
@@ -0,0 +1,144 @@
+import { useCallback, useMemo } from 'react';
+import { graphql, useFragment } from 'react-relay';
+
+import { useModalActions } from '~/contexts/modal/ModalContext';
+import { MintLinkButtonFragment$key } from '~/generated/MintLinkButtonFragment.graphql';
+import { FxHashLogoIcon } from '~/icons/FxHashLogoIcon';
+import { MintFunLogoIcon } from '~/icons/MintFunLogoIcon';
+import { ProhibitionLogoIcon } from '~/icons/ProhibitionLogoIcon';
+import { ZoraLogoIcon } from '~/icons/ZoraLogoIcon';
+import { contexts } from '~/shared/analytics/constants';
+import colors from '~/shared/theme/colors';
+import { MINT_LINK_DISABLED_CONTRACTS } from '~/shared/utils/communities';
+import {
+  getMintUrlWithReferrer,
+  MINT_LINK_CHAIN_ENABLED,
+} from '~/shared/utils/getMintUrlWithReferrer';
+
+import { Button, ButtonProps } from './core/Button/Button';
+import VerifyNavigationPopover from './core/GalleryLink/VerifyNavigationPopover';
+import { HStack } from './core/Spacer/Stack';
+
+type Props = {
+  tokenRef: MintLinkButtonFragment$key;
+  overwriteURL?: string;
+  referrerAddress?: string;
+  size?: 'sm' | 'md';
+} & ButtonProps;
+
+export function MintLinkButton({
+  overwriteURL,
+  referrerAddress,
+  tokenRef,
+  size = 'md',
+  variant = 'primary',
+  ...props
+}: Props) {
+  const token = useFragment(
+    graphql`
+      fragment MintLinkButtonFragment on Token {
+        definition {
+          community {
+            contract {
+              mintURL
+              contractAddress {
+                chain
+                address
+              }
+            }
+          }
+        }
+      }
+    `,
+    tokenRef
+  );
+
+  const { showModal } = useModalActions();
+
+  const tokenContractAddress =
+    token?.definition?.community?.contract?.contractAddress?.address ?? '';
+  const tokenChain = token?.definition?.community?.contract?.contractAddress?.chain ?? '';
+  const { url: mintURL, provider: mintProviderType } = getMintUrlWithReferrer(
+    overwriteURL ?? token?.definition?.community?.contract?.mintURL ?? '',
```

Comment:
want to make sure to do `overwriteURL ||` :)

Reply:
oh shit. Updated!